### PR TITLE
feature: local setup script

### PR DESCRIPTION
**Pull request by InoMurko**
A setup script that install all OS dependencies (currently tested only for linux distros).

## Changes

- ./bin/setup should make a local environment ready for bare metal development.

## Testing

checkout code + setup run + execute mix test.

Postgress installation is skipped because ... I think CircleCI doesn't allow creating users and roles manually.

### CODE REVIEW

**Comment by InoMurko** [comment]
@achiurizo does it make sense to buy circleci subscription so that we're able to test OSx as well?

File: bin/setup [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+install_darwin_deps(){
+    fancy_echo "Installing dependencies via brew"
+    brew bundle --file=- <<EOF
+brew "automake"
+brew "autoconf"
+brew "gmp"
+brew "libtool"
+brew "cmake"
```
[the remark this answers is not in the document]
Comment (InoMurko) [reply bin/setup]: whaaaat, I didn't know that!

File: bin/setup [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+#echo -e '\n. $HOME/.asdf/asdf.sh' >> ~/.bash_profile
+#echo -e '\n. $HOME/.asdf/completions/asdf.bash' >> ~/.bash_profile
+#source ~/.bash_profile
+#Linux and MacOS
+#asdf plugin-add elixir https://github.com/asdf-vm/asdf-elixir.git
+#asdf plugin-add erlang https://github.com/asdf-vm/asdf-erlang.git
+#asdf install erlang 21.0
+#asdf install elixir 1.8.2
+#asdf global elixir 1.8.2
+#asdf global erlang 21.0
```
[the remark this answers is not in the document]
Comment (InoMurko) [reply bin/setup]: Probably, I left it here so it's easier for others to install it.

File: bin/setup (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrhite) [reply bin/setup]: Interesting, didn't know that either! What distros does it work for?

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      install_linux_solc
+      install_linux_geth
+      if [ -z "$CIRLCECI" ]; then
+        echo "Skip Postgres setup, because CircleCI."
+      else
+        install_linux_postgres
+      fi
+      ;;
+    Darwin*)
+      install_darwin_deps
```
Comment (jrhite) [inline bin/setup]: No geth, solc, postgres setup for mac?

**Comment by jrhite** [comment]
Looks great man! This will be a big help and time-saver for folks.

File: bin/setup (hunk 3 above)
Comment (InoMurko) [reply bin/setup]: yeah... the problematic part is I have no idea how to test it. so rather then including something I'm not 100% sure it works, I left it open...

**Comment by jrhite** [comment]
Ah, one thing I forgot to mention/ask...

Do you want the setup script to append a `.bash_profile` file with the necessary ENV vars? These are some vars I needed to setup in my barebones ubuntu setup (my config is likely a bit dated, some things may have changed)...

```
export MIX_ENV=dev
export EXIT_PERIOD_SECONDS=45
export CONTRACT_EXCHANGER_URL=http://localhost:5000
export ETHEREUM_NETWORK=LOCAL_CHAIN
export ETHEREUM_RPC_URL=http://localhost:8545
export ETHEREUM_WS_URL=ws://localhost:8546
export CHILD_CHAIN_URL=http://localhost:9656
export ERL_CC_COOKIE=develop
export ERL_W_COOKIE=develop
export APP_ENV=local_development_child_chain
export DD_HOSTNAME=datadog
export DD_DISABLED=true
```

File: bin/setup (hunk 1 above)
[the remark this answers is not in the document]
Comment (InoMurko) [reply bin/setup]: so ok... I tried consolidating in using brew. the problem is that packages don't quite match:
erlang requirements like:
```
brew install libncurses5-dev
Error: No available formula with the name "libncurses5-dev"
```
```
asdf@localhost:~/elixir-omg$ brew install libssl-dev
Updating Homebrew...
Error: No available formula with the name "libssl-dev"
```

plus, there seems to be some confusion with package installations
```
asdf@localhost:~/elixir-omg$ asdf install erlang 21.0
Extracting source code
Building Erlang/OTP 21.0 (asdf_21.0), please wait...
WARNING: It appears that a required development package 'automake' is not installed.
WARNING: It appears that a required development package 'autoconf' is not installed.
^C
received sigint, cleaning up
asdf@localhost:~/elixir-omg$ brew install automake
Warning: automake 1.16.1_1 is already installed and up-to-date
To reinstall 1.16.1_1, run `brew reinstall automake`
```

[the remark this answers is not in the document]
Comment (achiurizo) [reply bin/setup]: @InoMurko ah that's unfortunate. I *think* you need to install that via a broader name like `brew install ncurses` ? But we can move forward without brew.

**Pull request merged**